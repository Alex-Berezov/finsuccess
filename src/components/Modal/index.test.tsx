import Modal from './index';
import { render, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event'

const ComponentForTest = () => {
  return (
    <h1>Render modal with component</h1>
  )
}

describe('Test for modal component', () => {
  const toggleModal = jest.fn()

  it('Render modal without component', () => {
    const ModalWithoutComponent = render(
      <Modal active={false} setActive={toggleModal} children={undefined} />
    )
    expect(ModalWithoutComponent).toMatchSnapshot();
  })

  it('Render modal with component', () => {
    const ModalWithComponent = render(
      <Modal active={true} setActive={toggleModal} children={<ComponentForTest />} />
    )
    expect(ModalWithComponent).toMatchSnapshot();
  })

  it('Check has class active', () => {
    const { container } = render(
      <Modal active={true} setActive={toggleModal} children={<ComponentForTest />} />
    )
    expect(container.firstChild).toHaveClass('active')
  })

  it('Close modal test', () => {
    const { getByTestId, queryByTestId } = render(
      <Modal active={true} setActive={toggleModal} children={<ComponentForTest />} />
    )
    userEvent.click(getByTestId('modal'))
    expect(toggleModal).toHaveBeenCalledTimes(1)
    expect(queryByTestId('modal')).not.toHaveClass('active', { exact: true })
  })
})