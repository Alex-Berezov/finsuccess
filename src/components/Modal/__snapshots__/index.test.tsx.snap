// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for modal component Render modal with component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal active"
        data-testid="modal"
      >
        <div
          class="modal__content active"
        >
          <h1>
            Render modal with component
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal active"
      data-testid="modal"
    >
      <div
        class="modal__content active"
      >
        <h1>
          Render modal with component
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test for modal component Render modal without component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="modal"
        data-testid="modal"
      >
        <div
          class="modal__content"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="modal"
      data-testid="modal"
    >
      <div
        class="modal__content"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
